### ISSUE TITLE
Improve Contribution Workflow

### ISSUE THREAD

**Issue by SachaG** [VulcanJS/Vulcan#1842]
Discuss!

**Comment by MHerszak** [comment]
https://github.com/blog/2111-issue-and-pull-request-templates

**Comment by justinr1234** [comment]
https://github.com/underarmour/further-review

**Comment by justinr1234** [comment]
CI to do the following on all PRs:

Circle CI and Travis both have free plans. It looks like Circle may be more performant. One thing to keep in mind is that builds could take a really long time if there's a lot of merges & prs.

- Run tests (if we had them)
- Run linting (very quick and easy)
- (this takes longer) Run a `meteor bundle` against the intended versions of `npm`, `node`, and `meteor` including a fresh `npm i` to make sure everything runs properly
- Run through a headless chrome as a smoke test (https://github.com/GoogleChrome/puppeteer)

Really if we could just verify a `npm run lint` I'd be pretty happy.

**Comment by luhagel** [comment]
[Here](https://gist.github.com/luhagel/b259ee464ffcded6a12d4169f13aab61) is a gist of my current CircleCI setup, runs pretty well and builds only take 1-2 minutes, so even when prs start piling up, we should be good to go.
Additionally, as an open source project, we get 3 free containers extra, which in theory allows 4 builds to run in parallel

**Comment by SachaG** [comment]
Some other guidelines: 

- All PRs should be made to the `devel` branch. 
- PRs to example packages should be made to the [Starter](https://github.com/VulcanJS/Vulcan-Starter) repo. 
- PRs need the approval of two contributor team members or one core team member before being merged. 
- Make separate PRs for separate features. 
- Only include commits relevant to your PR. 
- Do not include formatting/cosmetic changes, they make the diff harder to read. If you want to fix formatting, make a specific, separate PR. 
- If a PR is merged to `master` by mistake, just leave it be until we can discuss. 
- All PR code should be commented as much as possible. Better too many comments than too few. 
- Non-obvious bug fixes should include code comments referencing the GitHub issue/Stack Overflow question/documentation item/etc. explaining the fix. 
- Any new setting or callback in the PR code should be registered. 
- Any string in the PR code should be internationalized.

**Comment by justinr1234** [comment]
As part of the checklist for merging a PR, we should also see if it requires a documentation change.

**Comment by MathiasKandelborg** [comment]
Hi ! I got a bit sad when i saw the contributing.md and found this issue that hopefully adresses it. This template for creating [a good CONTRIBUTING.md](https://gist.github.com/PurpleBooth/b24679402957c63ec426) which serves as a great baseline.
